map/map_loader/src/pointcloud_map_loader/pointcloud_map_loader_node.cpp

```diff
@@ -89,8 +89,24 @@ std::map<std::string, PCDFileMetadata> PointCloudMapLoaderNode::getPCDMetadata(
   const std::string & pcd_metadata_path, const std::vector<std::string> & pcd_paths) const
 {
   if (fs::exists(pcd_metadata_path)) {
+    std::set<std::string> missing_pcds;
     auto pcd_metadata_dict = loadPCDMetadata(pcd_metadata_path);
-    pcd_metadata_dict = replaceWithAbsolutePath(pcd_metadata_dict, pcd_paths);
+
+    pcd_metadata_dict = replaceWithAbsolutePath(pcd_metadata_dict, pcd_paths, missing_pcds);
+
+    // Warning if some pcds are missing
+    if (!missing_pcds.empty()) {
+      std::ostringstream oss;
+
+      oss << "The following segment(s) are missing from the input PCDs: " << std::endl;
+
+      for (auto & fname : missing_pcds) {
+        oss << fname << std::endl;
+      }
+
+      RCLCPP_WARN_STREAM(get_logger(), oss.str());
```

Comment:
The situation where some pcd files are missing is very unusual, so I think just issuing a warning is insufficient.

When pcd files are missing, it's better to use `RCLCPP_ERROR` to log an error and throw a runtime_error or similar to exit the node. :pray:

Reply:
@KYabuuchi I added a throw runtime error to the end of the if clause.